index.js

```diff
@@ -219,6 +223,110 @@ let ContainerService = {
     });
   },
 
+  getTabs(args) {
+    return new Promise((resolve, reject) => {
+      if (!("userContextId" in args)) {
+        reject(null);
+        return;
+      }
+
+      let list = [];
+      let promises = [];
+
+      for (let tab of tabs) {
+        let xulTab = viewFor(tab);
+        let userContextId = parseInt(xulTab.getAttribute("usercontextid") || 0, 10);
```

Comment:
I notice this line (and the preceding line) is repeated a lot. Can it be refactored into a helper/utility function to reduce code duplication?